.github/workflows/deploy.yml

```diff
@@ -8,7 +8,200 @@ on:
+      run: |
+          echo "PROD_VERSION=$(aws cloudformation describe-stacks --stack-name prod-geospatial-data-lake-storage --query 'Stacks[].Tags[?Key==`Version`].Value[]' --output text)" | tee -a $GITHUB_ENV
+
+    - name: Checkout to current production version
+      run: git checkout ${{ env.PROD_VERSION }}
+
+    # dependencies installation
+    - name: Use Node for CDK deployment
+      uses: actions/setup-node@v2.1.5
+      with:
+        node-version: 14
+        registry-url: https://registry.npmjs.org
+
+    - name: Cache Node.js packages
+      uses: actions/cache@v2.1.4
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-
+
+    - name: Install Node.js dependencies
+      run: npm ci --production
+
+    - name: Add local Node packages to PATH
+      run: echo "./node_modules/.bin:$PATH" >> $GITHUB_PATH
+
+    - name: Use Python ${{ matrix.python }}
+      uses: actions/setup-python@v2.2.1
+      with:
+        python-version: ${{ matrix.python }}
+
+    - name: Cache pip
+      uses: actions/cache@v2.1.4
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ matrix.python }}-${{ hashFiles('./poetry.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-${{ matrix.python }}-
+
+    - name: Install Python dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install poetry
+        python -m poetry install \
+            --extras='cdk check_files_checksums check_stac_metadata content_iterator datasets dataset_versions import_dataset import_status validation_summary'
+
+    - name: Print CDK version
+      run: poetry run cdk --version
+
+    # deployment
+    - name: Configure CI AWS credentials
+      if: >
+        startsWith(github.ref, 'refs/heads/release-') || startsWith(github.ref, 'refs/tags/release-')
+        && github.repository == 'linz/geospatial-data-lake'
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: ap-southeast-2
+        mask-aws-account-id: true
+        role-to-assume: ${{ secrets.AWS_ASSUME_ROLE_CI }}
+
+    - name: Set unique deployment environment type variable
+      run: |
+        DEPLOY_ENV="ci${GITHUB_RUN_ID}"
+        echo "DEPLOY_ENV=$DEPLOY_ENV" | tee -a $GITHUB_ENV
+
+    - name: Deploy copy of production AWS stacks in to CI environment
+      if: >
+        startsWith(github.ref, 'refs/heads/release-') || startsWith(github.ref, 'refs/tags/release-')
+        && github.repository == 'linz/geospatial-data-lake'
+      run: |
+        poetry run cdk bootstrap aws://unknown-account/ap-southeast-2
+        poetry run cdk deploy --all --require-approval never
+
+
+  upgrade-copy-of-prod:
+    needs: deploy-copy-of-prod
```

Comment:
Is this intentional? Doesn't this mean that we can't upgrade without running deploy *in the same pipeline* first?

Reply:
This is testing of prod upgrade done on prod copy in CI environment